tests/test_estimate_pz_task.py

```diff
@@ -0,0 +1,239 @@
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Unit tests for meaz_pz
+"""
+
+import os
+import unittest
+import subprocess
+
+import qp
+from rail.core.model import Model as PZModel
+
+from lsst.daf.butler import (
+    Butler,
+    DataCoordinate,
+    DatasetType,
+    DatasetRef,
+    DimensionGroup,
+    DimensionUniverse,
+    FileDataset,
+)
+from lsst.meas.pz.estimate_pz_task import EstimatePZTask
+from lsst.meas.pz.estimate_pz_task_trainz import EstimatePZTrainZTask
+from lsst.meas.pz.estimate_pz_task_knn import EstimatePZKNNTask
+
+PIPELINES_DIR = os.path.join(os.path.dirname(__file__), "..", "pipelines")
+TEST_DIR = os.path.abspath(os.path.dirname(__file__))
+CI_HSC_GEN3_DIR = os.environ.get("CI_HSC_GEN3_DIR", None)
+DAF_BUTLER_REPOSITORY_INDEX = os.environ.get("DAF_BUTLER_REPOSITORY_INDEX", None)
+IS_S3DF = DAF_BUTLER_REPOSITORY_INDEX == "/sdf/group/rubin/shared/data-repos.yaml"
+
+
+class MeasPzTasksTestCase(unittest.TestCase):
+
+    dim_universe = DimensionUniverse()
+
+    objectTable_dimension_group = DimensionGroup(
+        dim_universe,
+        ["skymap", "tract", "patch"],
+    )
+
+    objectTable_datasetType = DatasetType(
+        "objectTable",
+        dimensions=objectTable_dimension_group,
+        storageClass="DataFrame",
+    )
+
+    pzModel_dimension_group = DimensionGroup(
+        dim_universe,
+        ["instrument"],
+    )
+
+    pzModel_trainz_datasetType = DatasetType(
+        "pzModel_trainz",
+        dimensions=pzModel_dimension_group,
+        storageClass="PZModel",
+        isCalibration=True,
+    )
+
+    pzModel_knn_datasetType = DatasetType(
+        "pzModel_knn",
+        dimensions=pzModel_dimension_group,
+        storageClass="PZModel",
+        isCalibration=True,
+    )
+
+    def makeButler_ci_hsc(self, **kwargs):
+        butler = Butler(
+            os.path.abspath(os.path.join(CI_HSC_GEN3_DIR, "DATA")), **kwargs
+        )
+        return butler
+
+    def makeButler_repo_dc2(self, **kwargs):
+        butler = Butler(
+            "/repo/dc2",
+            collections=[
+                "2.2i/runs/test-med-1/w_2024_16/DM-43972/step3/group1/w00_000"
```

Comment:
This only runs if you explicitly run py.test at s3df.   We could imagine having a specific set of test collections or a test repo at s3df to avoid these sorts of issues.   For now it is just nice to be able to do this as part of running py.test

Reply:
Also worth noting that I _think_ this doesn't put anything into the DB, so we can run it multiple times safely.